pkg/controller/cloud/node_controller.go

```diff
@@ -239,9 +246,28 @@ func (cnc *CloudNodeController) MonitorNode() {
 		// from the cloud provider. If node cannot be found in cloudprovider, then delete the node immediately
 		if currentReadyCondition != nil {
 			if currentReadyCondition.Status != v1.ConditionTrue {
+				// we need to check this first to get taint working in similar in all cloudproviders
```

Comment:
can we have some tests for this ?

Reply:
I can add those when i have time. Hopefully today/tomorrow

Reply:
@yastij tests added now